tests/checkout/index.c

```diff
@@ -175,35 +193,45 @@ void test_checkout_index__honor_coresymlinks_default(void)
 		cl_assert_equal_s(link_data, "new.txt");
 		check_file_contents("./symlink/link_to_new.txt", "my new file\n");
 	}
-#endif
 
 	cl_fixture_cleanup("symlink");
 }
 
+void test_checkout_index__coresymlinks_set_to_true_fails_when_unsupported(void)
+{
+	git_checkout_options opts = GIT_CHECKOUT_OPTIONS_INIT;
+
+	if (supports_symlinks("testrepo")) {
+		cl_skip();
+	}
+
+	cl_repo_set_bool(g_repo, "core.symlinks", true);
+
+	opts.checkout_strategy = GIT_CHECKOUT_SAFE | GIT_CHECKOUT_RECREATE_MISSING;
+	cl_git_fail(git_checkout_index(g_repo, NULL, &opts));
+}
+
 void test_checkout_index__honor_coresymlinks_setting_set_to_true(void)
 {
 	git_checkout_options opts = GIT_CHECKOUT_OPTIONS_INIT;
+	char link_data[1024];
+	size_t link_size = 1024;
```

Comment:
We could use `GIT_PATH_MAX` instead here to make this a bit less magical. But it's not like it'd matter much anyway, so feel free to ignore this remark

Reply:
Agreed and done.